packages/blog/src/components/common/SearchPosts/SearchPosts.tsx

```diff
@@ -13,20 +14,18 @@ type SearchPostsProps = {
 
 export function SearchPosts({ inputTitle, inputPlaceholder }: SearchPostsProps) {
   const [searchValue, setSearchValue] = useState<string>('')
-  const debouncedValue = useDebounce<string>(searchValue, 500)
-  const [postResults, setPostResults] = useState<AlgoliaPost[]>([])
+  const debouncedValue = useDebounce<string>(searchValue, 300)
 
-  useEffect(() => {
-    if (debouncedValue.length) {
-      getAlgoliaPosts({ query: debouncedValue })
-        .then(({ hits }) => {
-          setPostResults(hits)
-        })
-        .catch(err => {
-          console.log(err)
-        })
-    }
-  }, [debouncedValue])
+  const { data: postResults } = useQuery(
+    ['algoliaPosts', { debouncedValue }],
+    async () => getAlgoliaPosts({ query: debouncedValue }).then(({ hits }) => hits.filter(hit => hit.publishedAt)),
+    {
+      enabled: debouncedValue.length > 0,
+      refetchOnWindowFocus: false,
+      staleTime: 30000,
+      onError: err => console.log(err),
+    },
+  )
```

Comment:
This is not a big deal, but following this logic, the usual case will require 3 renders:
- set the value
- set the denounced value after 300ms
- set the date once the query finishes

We can improve it to two renders by using a denounced call instead of a denounced value. If you set the `enabled` option to false, and then debounce the `refetch` function returned by `useQuery`, you should be able to do it on demand.

But as I said, that's not a huge improvement so leaving this as-is is also a good option 🤓